### PR TITLE
Add User, Group, StandardOutput and StandardError services.

### PR DESCRIPTION
**Pull request by bschonec**
#### Pull Request (PR) description
I've added a few units and services that I need and were missing from this project.

### CODE REVIEW

**Comment by bschonec** [comment]
Done

**Comment by kenyon** [comment]
Now a rubocop violation, not sure where this is coming from.

```
Error: Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body end. ([https://rubystyle.guide#empty-lines-around-bodies](https://rubystyle.guide/#empty-lines-around-bodies))
Error: Process completed with exit code 1.
```

**Comment by hboetes** [comment]
Much appreciated, @jhoblitt !

File: types/unit/service.pp [hunk 1]
```diff
@@ -39,5 +39,10 @@
     Optional['OOMScoreAdjust']            => Integer[-1000,1000],
     Optional['Environment']               => String,
     Optional['EnvironmentFile']           => Variant[Stdlib::Unixpath,Pattern[/-\/.*/]],
+    Optional['User']                      => String,
+    Optional['Group']                     => String,
+    Optional['StandardOutput']            => String,
+    Optional['StandardError']             => String,
```
Comment (bastelfreak) [inline types/unit/service.pp:45]: Should we restrict this and don't allow empty strings?
```suggestion
    Optional['User']                      => String[1],
    Optional['Group']                     => String[1],
    Optional['StandardOutput']            => String[1],
    Optional['StandardError']             => String[1],
```

File: types/unit/service.pp [hunk 2]
```diff
@@ -39,5 +39,10 @@
...
+    Optional['WorkingDirectory']          => Stdlib::Unixpath,
```
Comment (kenyon) [inline types/unit/service.pp:46]: `Stdlib::Unixpath` actually looks too restrictive: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=

Also `~` and paths starting with `-` are allowed.

**Comment by ekohl** [comment]
@bschonec I could also use this functionality (already opened #321). Any chance you can revisit this and address the feedback?

**Comment by bschonec** [comment]
> @bschonec I could also use this functionality (already opened #321). Any chance you can revisit this and address the feedback?

I'll try.  I've got very, very little experience submitting good code to the Vox Pupuli group.

**Comment by traylenator** [comment]
Looks fine with a re-base , obviously user and group at least added already so just add the others.

**Comment by bschonec** [comment]
Why was this closed?

**Comment by ekohl** [comment]
Because #353 implemented it.

**Comment by traylenator** [comment]
@bschonec thanks for the patch - I used it in #353 fixed for the things that were already added.

**Comment by bschonec** [comment]
Naw, you did all the work...thanks for implementing it and explaining how my issue was closed.

**Pull request closed without merging**